File: store_credit/store_credit.rb
def _solve credit, items
  head, *rest = items 
  if found = rest.find {|p| head[0] + p[0] == credit}
    [head, found]
  else
    _solve credit, rest
  end   
end

def solve credit, items
  indexed = items.zip((1..items.length).to_a)
  cand = indexed # indexed.select {|p| p[0] < credit}.sort {|p| p[0]}.reverse 
  values = _solve(credit, cand)
  values.collect(&:last).sort
end

N = gets.to_i
no = 0

while (c = gets) do 
  c = c.to_i
  i, items = gets.to_i, gets.split.collect(&:to_i)
  puts "Case ##{no += 1}: #{solve(c, items).join(' ')}"  
end

File: store_credit/store_credit.js
// JavaScript file for running on node.js

function solve(credit, items) {
	var head = items.shift();
	var found;
	if (items.some(function(e) { found = e; return e[0] + head[0] == credit })) {
		return [head, found];
	}
	return solve(credit, items);
}

var buffer = "";
process.stdin.on('data', function(chunk) {
	buffer = buffer + chunk;
});

process.stdin.on('end', function() {
	var lines = buffer.split("\n");
	var N = parseInt(lines.shift());
	for (n = 1; n <= N; n++) {
		var credit = parseInt(lines.shift());
		lines.shift();
		var index = 1;
		var values = lines.shift().split(" ").map(function(e) { return [parseInt(e), index++] });
		var r = solve(credit, values).map(function(e) { return parseInt(e[1]) }).sort(function(a, b) {
			return (a > b) ? 1 : -1;
		});
		console.log("Case #" + n + ": " + r.join(" "));		
	}
});

process.stdin.resume();